### ISSUE TITLE
Don't override exported variable that have not being setted

### ISSUE THREAD

**Issue by thiagola92** [coppolaemilio/dialogic#1966]
**Is your feature request related to a problem? Please describe.**
I have a custom portrait (`extends DialogicPortrait`) and I want to use `@export` to hold information about other node in the scene.  

![image](https://github.com/coppolaemilio/dialogic/assets/9352894/83cd97c5-ef6d-4d89-aa75-0d799c11265a)

Dialogic goes through all exported variables and save the default values, to later override exported variables.  

The problem is that the default value can be `null` at this step, which would make it later override with `null`.  

```gdscript
# Default value will be null
@export var animated_sprite: AnimatedSprite2D

# Default value will be "default"
@export var animation: String = "default"
```

It seems to be doing:
- Dialogic get default values
- Godot set scene values
- Dialogic set default values

**Describe the solution you'd like**
Options:
1. Dialogic being explicit that exported variables of type `TYPE_OBJECT` will be set to null.
2. Dialogic not overriding my exported variables with `null`.

**Describe alternatives you've considered**
Adapt script to not use `@export` (this would means losing a feature of Godot when dealing with Dialogic).  

**Additional context**

**Solution 1**:
Like `ColisionShape2D`, an explicit warning should be telling the user that using a exported variable of the type `TYPE_OBJECT` will override the value to `null`.  

![image](https://github.com/coppolaemilio/dialogic/assets/9352894/cf419186-e27e-416d-9352-977d37ffe598)

**Solution 2.1**:
Don't save default value of unsupported types (like `TYPE_OBJECT`).  

https://github.com/coppolaemilio/dialogic/blob/3e90442dd02ff325a70adf3105d2d7de23ea6a7e/addons/dialogic/Other/DialogicUtil.gd#L265-L267  

```gdscript
	for i in property_info:
		if i['usage'] & PROPERTY_USAGE_EDITOR and i['type'] != TYPE_OBJECT:
			defaults[i['name']] = node.get(i['name'])
```

**Solution 2.2**:  
Don't save default values when `null`.  

```gdscript
	for i in property_info:
		if i['usage'] & PROPERTY_USAGE_EDITOR and node.get(i['name']) != null:
			defaults[i['name']] = node.get(i['name'])
```

**Comment by Jowan-Spooner** [comment]
@thiagola92 Thanks for this report, it does in fact seem like annoying behaviour. I'm a bit annoyed that these objects are not correctly recognized as the defaults. I'm gonna try investigating this a bit more.

However for your exact problem it might be easier to solve it by adding an ignored group like we already have for layout layers (these can put exports in a "private" group and then dialogic won't bother with them). I assume you didn't actually want to change the animated sprite via the portrait settings, but I might be wrong.

**Comment by thiagola92** [comment]
> I assume you didn't actually want to change the animated sprite via the portrait settings, but I might be wrong.

Actually I did, I'm new to Dialogic and my intention was not to create many scenes and reuse the `AnimatedSprite2D` animations.  

[Screencast from 2023-12-22 09-30-22.webm](https://github.com/coppolaemilio/dialogic/assets/9352894/271578c2-5ee5-4d4b-81ba-d36a2c7d12fe)

> However for your exact problem it might be easier to solve it by adding an ignored group like we already have for layout layers (these can put exports in a "private" group and then dialogic won't bother with them).

I will have to look into it the documentation because I have no idea about this.

**Comment by Jowan-Spooner** [comment]
fair enough. As a work-around you can make an exported variable to a path of the animated sprite resource and then load it in _ready().

Something like 
```
@export_file("*.tres") var anim_sprite_file := "res://my_folder/animated_sprite.tres"
var animated_sprite :AnimatedSprite = null

func _ready():
     animated_sprite = load(anim_sprite_file)
```
---
> I will have to look into it the documentation because I have no idea about this.

This only exists for layouts right now. I meant we could add the same thing to portraits, but they don't have it yet (I think).

**Comment by thiagola92** [comment]
Thank you, I will rewrite this way!
